pkg/controller/stack/stack_controller.go

```diff
@@ -209,6 +209,20 @@ func (r *ReconcileStack) Reconcile(request reconcile.Request) (reconcile.Result,
 		}
 	}
 
+	// If a branch is specified, then track changes to the branch.
+	trackBranch := len(sess.stack.Branch) > 0
```

Comment:
Any reason why the change to disable branch tracking if an explicit commit is specified?

Reply:
I realized that `branch` and `commit` are mutually exclusive, so we should only need to check for the presence of `branch`.

https://github.com/pulumi/pulumi-kubernetes-operator/blob/597053a08e0081dfc4a4e9f2877467d28de91bb9/pkg/apis/pulumi/v1alpha1/stack_types.go#L96-L101